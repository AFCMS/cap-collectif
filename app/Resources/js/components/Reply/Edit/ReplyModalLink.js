// @flow
import React, { Fragment } from 'react';
import { FormattedMessage, FormattedDate } from 'react-intl';
import { ListGroupItem, Button } from 'react-bootstrap';
import { createFragmentContainer, graphql } from 'react-relay';
import moment from 'moment';
import type { ReplyModalLink_reply } from './__generated__/ReplyModalLink_reply.graphql';
import UpdateReplyModal from './UpdateReplyModal';
import ReplyDraftLabel from './ReplyDraftLabel';
import UnpublishedLabel from '../../Publishable/UnpublishedLabel';
import DeleteReplyModal from '../Delete/DeleteReplyModal';

type Props = {
  reply: ReplyModalLink_reply,
  questionnaire: Object,
};

type CurrentOpenModal = 'edit' | 'delete' | null;

type State = {
  currentOpenModal: CurrentOpenModal,
};

export class ReplyModalLink extends React.Component<Props, State> {
  state = {
    currentOpenModal: null,
  };

  showModal = (modal: CurrentOpenModal) => {
    this.setState({ currentOpenModal: modal });
  };

  hideModal = () => {
    this.setState({ currentOpenModal: null });
  };

  render() {
    const { reply, questionnaire } = this.props;

    return (
      <Fragment>
        <ListGroupItem className="reply align-items-center" id={`reply-link-${reply.id}`}>
          <div>
            <FormattedMessage
              id="reply.show.link"
              values={{
                date: (
                  <FormattedDate
                    value={moment(reply.publishedAt ? reply.publishedAt : reply.createdAt)}
                    day="numeric"
                    month="long"
                    year="numeric"
                  />
                ),
                time: (
                  <FormattedDate value={moment(reply.createdAt)} hour="numeric" minute="numeric" />
                ),
              }}
            />
            {reply.private && (
              <span>
                {' '}
                <FormattedMessage id="reply.private" />
              </span>
            )}
            {/* $FlowFixMe $refType */}
            <ReplyDraftLabel status={reply.publicationStatus} />
            <UnpublishedLabel publishable={reply} />
          </div>
          <div>
            <Button
              className="mr-10 reply__update-btn"
              bsStyle="warning"
              onClick={() => this.showModal('edit')}>
              <FormattedMessage id="glodal.edit" />
            </Button>
            <Button bsStyle="danger reply__delete-btn" onClick={() => this.showModal('delete')}>
              <FormattedMessage id="glodal.delete" />
            </Button>
          </div>
        </ListGroupItem>
        {/* $FlowFixMe $refType */}
        <UpdateReplyModal
          show={this.state.currentOpenModal === 'edit'}
          onClose={this.hideModal}
          reply={reply}
          questionnaire={questionnaire}
        />
        {/* $FlowFixMe $refType */}
        <DeleteReplyModal
          reply={reply}
          show={this.state.currentOpenModal === 'delete'}
          onClose={this.hideModal}
        />
      </Fragment>
    );
  }
}

export default createFragmentContainer(ReplyModalLink, {
  reply: graphql`
    fragment ReplyModalLink_reply on Reply {
      createdAt
      publishedAt
      id
      private
      publicationStatus
      ...UpdateReplyModal_reply
      ...DeleteReplyModal_reply
      ...UnpublishedLabel_publishable
    }
  `,
});
