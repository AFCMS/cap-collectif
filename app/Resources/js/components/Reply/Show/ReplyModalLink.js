// @flow
import React, { Fragment } from 'react';
import { FormattedMessage, FormattedDate } from 'react-intl';
import { ListGroupItem } from 'react-bootstrap';
import { createFragmentContainer, graphql } from 'react-relay';
import moment from 'moment';
import type { ReplyModalLink_reply } from './__generated__/ReplyModalLink_reply.graphql';
import ShowReplyModal from './ShowReplyModal';

type Props = {
  reply: ReplyModalLink_reply,
  questionnaire: Object,
};

type State = {
  showModal: boolean,
};

export class ReplyModalLink extends React.Component<Props, State> {
  state = {
    showModal: false,
  };

  showModal = () => {
    this.setState({
      showModal: true,
    });
  };

  hideModal = () => {
    this.setState({
      showModal: false,
    });
  };

  render() {
    const { reply, questionnaire } = this.props;

    return (
      <Fragment>
        <ListGroupItem className="reply" id={`reply-link-${reply.id}`} onClick={this.showModal}>
          <FormattedMessage
            id="reply.show.link"
            values={{
              date: (
                <FormattedDate
                  value={moment(reply.createdAt)}
                  day="numeric"
                  month="long"
                  year="numeric"
                />
              ),

              time: (
                <FormattedDate value={moment(reply.createdAt)} hour="numeric" minute="numeric" />
              ),
            }}
          />
          {reply.private && (
            <span>
              {' '}
              <FormattedMessage id="reply.private" />
            </span>
          )}
        </ListGroupItem>
        {/* $FlowFixMe $refType */}
        <ShowReplyModal
          show={this.state.showModal}
          onClose={this.hideModal}
          reply={reply}
          questionnaire={questionnaire}
        />
      </Fragment>
    );
  }
}

export default createFragmentContainer(ReplyModalLink, {
  reply: graphql`
    fragment ReplyModalLink_reply on Reply {
      createdAt
      id
      private
      ...ShowReplyModal_reply
    }
  `,
});
