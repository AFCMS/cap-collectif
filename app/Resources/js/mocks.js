// @flow
/* eslint-env jest */
import { type FormProps } from 'redux-form';
import { type IntlShape } from 'react-intl';
import type { RelayPaginationProp, RelayRefetchProp } from 'react-relay';

export const $refType: any = null;
export const $fragmentRefs: any = null;

export const intlMock: IntlShape = {
  locale: 'fr-FR',
  formats: {},
  messages: {},
  now: () => 0,
  formatHTMLMessage: message => String(message),
  formatPlural: message => String(message),
  formatNumber: message => String(message),
  formatRelative: message => String(message),
  formatTime: message => String(message),
  formatDate: message => String(message),
  formatMessage: message => String(message.id),
};

export const formMock: FormProps = {
  anyTouched: false,
  array: {
    insert: jest.fn(),
    move: jest.fn(),
    pop: jest.fn(),
    push: jest.fn(),
    remove: jest.fn(),
    removeAll: jest.fn(),
    shift: jest.fn(),
    splice: jest.fn(),
    swap: jest.fn(),
    unshift: jest.fn(),
  },
  asyncValidate: jest.fn(),
  asyncValidating: false,
  autofill: jest.fn(),
  blur: jest.fn(),
  change: jest.fn(),
  clearAsyncError: jest.fn(),
  clearSubmit: jest.fn(),
  destroy: jest.fn(),
  dirty: false,
  dispatch: jest.fn(),
  error: null,
  form: 'formName',
  handleSubmit: jest.fn(),
  initialize: jest.fn(),
  initialized: true,
  initialValues: {},
  invalid: false,
  pristine: true,
  reset: jest.fn(),
  submitting: false,
  submitFailed: false,
  submitSucceeded: false,
  touch: jest.fn(),
  untouch: jest.fn(),
  valid: true,
  warning: null,
};

export const relayPaginationMock: RelayPaginationProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  hasMore: () => false,
  isLoading: () => false,
  loadMore: jest.fn(),
  refetchConnection: jest.fn(),
};

export const relayRefetchMock: RelayRefetchProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  refetch: jest.fn(),
};
