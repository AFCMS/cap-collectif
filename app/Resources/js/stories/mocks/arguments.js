// @flow
import { author } from './users';

export const argument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const argumentsData = [
  { ...argument, user: { ...argument.user, vip: true } },
  { ...argument },
  { ...argument, trashedStatus: 'INVISIBLE' },
  { ...argument },
];
