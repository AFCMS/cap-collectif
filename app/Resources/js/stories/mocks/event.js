// @flow
import { author } from './users';

export const event = {
  title: 'PHPTourDuTurfu',
  url: '#',
  image: {
    url: 'https://picsum.photos/300/400',
    alt: 'event picture',
  },
  date: {
    start: '2030-03-10 00:00:00',
    end: '2032-06-15 00:00:00',
  },
  location: {
    place: 'Tour Eiffel',
    zip: '75007',
    city: 'Paris',
  },
  category: 'Immobilier',

  user: author,
};
