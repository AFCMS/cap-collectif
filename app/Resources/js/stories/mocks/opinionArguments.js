// @flow
import { author } from './users';

export const opinionArgument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const opinionArguments = [
  { ...opinionArgument, user: { ...opinionArgument.user, vip: true } },
  { ...opinionArgument },
  { ...opinionArgument, trashedStatus: 'INVISIBLE' },
  { ...opinionArgument },
];
