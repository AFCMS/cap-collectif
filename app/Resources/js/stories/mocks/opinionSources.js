// @flow
import { author } from './users';

export const opinionSource = {
  url: 'https://ui.cap-collectif.com/',
  user: author,
  createdAt: ' • 1 mars 2016',
  publishedAt: ' • 1 mars 2016',
  title: 'Source title',
  category: 'Politique',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 2 },
};

export const opinionSources = [
  { ...opinionSource },
  { ...opinionSource },
  { ...opinionSource, user: { ...opinionSource.user, vip: true } },
  { ...opinionSource },
  { ...opinionSource, category: null },
];
