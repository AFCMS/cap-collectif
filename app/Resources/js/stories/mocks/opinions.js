// @flow
import { author } from './users';

export const opinion = {
  url: 'https://ui.cap-collectif.com/',
  title: 'Opinion',
  user: author,
  createdAt: ' • 1 mars 2018',
  pinned: true,
  publishedAt: ' • 1 mars 2018',
  votes: { totalCount: 4 },
  versions: { totalCount: 3 },
  arguments: { totalCount: 4 },
  sources: { totalCount: 0 },
  votesMitige: { totalCount: 1 },
  votesNok: { totalCount: 1 },
  votesOk: { totalCount: 2 },
};

export const opinions = [
  { ...opinion },
  { ...opinion, user: { ...opinion.user, vip: true } },
  { ...opinion, votes: { totalCount: 0 } },
];
