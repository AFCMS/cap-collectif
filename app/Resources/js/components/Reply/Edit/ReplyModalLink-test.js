// @flow
/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import { ReplyModalLink } from './ReplyModalLink';
import { $refType, $fragmentRefs } from '../../../mocks';

describe('<ReplyModalLink />', () => {
  const reply = {
    $refType,
    createdAt: '2016-03-01 10:11:12',
    publishedAt: '2016-04-25 14:33:17',
    id: 'replay1',
    private: false,
    draft: false,
    viewerCanUpdate: true,
    viewerCanDelete: true,
    $fragmentRefs,
  };
  const notContribuableReply = {
    ...reply,
    viewerCanUpdate: false,
    viewerCanDelete: false,
  };

  it('render a reply in a contribuable questionnaire', () => {
    const wrapper = shallow(<ReplyModalLink reply={reply} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render a draft reply in a contribuable questionnaire', () => {
    const wrapper = shallow(
      <ReplyModalLink reply={{ ...reply, draft: true, publishedAt: null }} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('render a private reply in a contribuable questionnaire', () => {
    const wrapper = shallow(<ReplyModalLink reply={{ ...reply, private: true }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render a reply in a closed questionnaire', () => {
    const wrapper = shallow(<ReplyModalLink reply={notContribuableReply} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render a draft reply in a closed questionnaire', () => {
    const wrapper = shallow(
      <ReplyModalLink reply={{ ...notContribuableReply, draft: true, publishedAt: null }} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
