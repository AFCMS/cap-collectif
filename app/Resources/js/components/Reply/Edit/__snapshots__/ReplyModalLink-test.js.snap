// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReplyModalLink /> render a draft reply in a closed questionnaire 1`] = `
<Fragment>
  <ListGroupItem
    bsClass="list-group-item"
    className="reply align-items-center"
    id="reply-link-replay1"
    listItem={false}
  >
    <div>
      <FormattedMessage
        id="reply.show.link"
        values={
          Object {
            "date": <FormattedDate
              day="numeric"
              month="long"
              value={null}
              year="numeric"
            />,
            "time": <FormattedDate
              hour="numeric"
              minute="numeric"
              value={null}
            />,
          }
        }
      />
      <ReplyDraftLabel
        draft={true}
      />
      <ForwardRef(Relay(Connect(UnpublishedLabel)))
        publishable={
          Object {
            "$fragmentRefs": null,
            "$refType": null,
            "createdAt": "",
            "draft": true,
            "id": "replay1",
            "private": false,
            "publishedAt": "",
            "viewerCanDelete": false,
            "viewerCanUpdate": false,
          }
        }
      />
    </div>
    <div>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="primary"
        className="mr-10 reply__update-btn mt-5"
        disabled={false}
        onClick={[Function]}
      >
        <FormattedMessage
          id="global.see"
          values={Object {}}
        />
      </Button>
    </div>
  </ListGroupItem>
  <ForwardRef(Relay(UpdateReplyModal))
    onClose={[Function]}
    reply={
      Object {
        "$fragmentRefs": null,
        "$refType": null,
        "createdAt": "",
        "draft": true,
        "id": "replay1",
        "private": false,
        "publishedAt": "",
        "viewerCanDelete": false,
        "viewerCanUpdate": false,
      }
    }
    show={false}
  />
  <ForwardRef(Relay(DeleteReplyModal))
    onClose={[Function]}
    reply={
      Object {
        "$fragmentRefs": null,
        "$refType": null,
        "createdAt": "",
        "draft": true,
        "id": "replay1",
        "private": false,
        "publishedAt": "",
        "viewerCanDelete": false,
        "viewerCanUpdate": false,
      }
    }
    show={false}
  />
</Fragment>
`;

exports[`<ReplyModalLink /> render a draft reply in a contribuable questionnaire 1`] = `
<Fragment>
  <ListGroupItem
    bsClass="list-group-item"
    className="reply align-items-center"
    id="reply-link-replay1"
    listItem={false}
  >
    <div>
      <FormattedMessage
        id="reply.show.link"
        values={
          Object {
            "date": <FormattedDate
              day="numeric"
              month="long"
              value={null}
              year="numeric"
            />,
            "time": <FormattedDate
              hour="numeric"
              minute="numeric"
              value={null}
            />,
          }
        }
      />
      <ReplyDraftLabel
        draft={true}
      />
      <ForwardRef(Relay(Connect(UnpublishedLabel)))
        publishable={
          Object {
            "$fragmentRefs": null,
            "$refType": null,
            "createdAt": "",
            "draft": true,
            "id": "replay1",
            "private": false,
            "publishedAt": "",
            "viewerCanDelete": true,
            "viewerCanUpdate": true,
          }
        }
      />
    </div>
    <div>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="warning"
        className="mr-10 reply__update-btn mt-5"
        disabled={false}
        onClick={[Function]}
      >
        <FormattedMessage
          id="glodal.edit"
          values={Object {}}
        />
      </Button>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="danger"
        className="reply__delete-btn mt-5"
        disabled={false}
        onClick={[Function]}
      >
        <FormattedMessage
          id="glodal.delete"
          values={Object {}}
        />
      </Button>
    </div>
  </ListGroupItem>
  <ForwardRef(Relay(UpdateReplyModal))
    onClose={[Function]}
    reply={
      Object {
        "$fragmentRefs": null,
        "$refType": null,
        "createdAt": "",
        "draft": true,
        "id": "replay1",
        "private": false,
        "publishedAt": "",
        "viewerCanDelete": true,
        "viewerCanUpdate": true,
      }
    }
    show={false}
  />
  <ForwardRef(Relay(DeleteReplyModal))
    onClose={[Function]}
    reply={
      Object {
        "$fragmentRefs": null,
        "$refType": null,
        "createdAt": "",
        "draft": true,
        "id": "replay1",
        "private": false,
        "publishedAt": "",
        "viewerCanDelete": true,
        "viewerCanUpdate": true,
      }
    }
    show={false}
  />
</Fragment>
`;

exports[`<ReplyModalLink /> render a private reply in a contribuable questionnaire 1`] = `
<Fragment>
  <ListGroupItem
    bsClass="list-group-item"
    className="reply align-items-center"
    id="reply-link-replay1"
    listItem={false}
  >
    <div>
      <FormattedMessage
        id="reply.show.link"
        values={
          Object {
            "date": <FormattedDate
              day="numeric"
              month="long"
              value={null}
              year="numeric"
            />,
            "time": <FormattedDate
              hour="numeric"
              minute="numeric"
              value={null}
            />,
          }
        }
      />
      <span>
         
        <FormattedMessage
          id="reply.private"
          values={Object {}}
        />
      </span>
      <ReplyDraftLabel
        draft={false}
      />
      <ForwardRef(Relay(Connect(UnpublishedLabel)))
        publishable={
          Object {
            "$fragmentRefs": null,
            "$refType": null,
            "createdAt": "",
            "draft": false,
            "id": "replay1",
            "private": true,
            "publishedAt": "",
            "viewerCanDelete": true,
            "viewerCanUpdate": true,
          }
        }
      />
    </div>
    <div>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="warning"
        className="mr-10 reply__update-btn mt-5"
        disabled={false}
        onClick={[Function]}
      >
        <FormattedMessage
          id="glodal.edit"
          values={Object {}}
        />
      </Button>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="danger"
        className="reply__delete-btn mt-5"
        disabled={false}
        onClick={[Function]}
      >
        <FormattedMessage
          id="glodal.delete"
          values={Object {}}
        />
      </Button>
    </div>
  </ListGroupItem>
  <ForwardRef(Relay(UpdateReplyModal))
    onClose={[Function]}
    reply={
      Object {
        "$fragmentRefs": null,
        "$refType": null,
        "createdAt": "",
        "draft": false,
        "id": "replay1",
        "private": true,
        "publishedAt": "",
        "viewerCanDelete": true,
        "viewerCanUpdate": true,
      }
    }
    show={false}
  />
  <ForwardRef(Relay(DeleteReplyModal))
    onClose={[Function]}
    reply={
      Object {
        "$fragmentRefs": null,
        "$refType": null,
        "createdAt": "",
        "draft": false,
        "id": "replay1",
        "private": true,
        "publishedAt": "",
        "viewerCanDelete": true,
        "viewerCanUpdate": true,
      }
    }
    show={false}
  />
</Fragment>
`;

exports[`<ReplyModalLink /> render a reply in a closed questionnaire 1`] = `
<Fragment>
  <ListGroupItem
    bsClass="list-group-item"
    className="reply align-items-center"
    id="reply-link-replay1"
    listItem={false}
  >
    <div>
      <FormattedMessage
        id="reply.show.link"
        values={
          Object {
            "date": <FormattedDate
              day="numeric"
              month="long"
              value={null}
              year="numeric"
            />,
            "time": <FormattedDate
              hour="numeric"
              minute="numeric"
              value={null}
            />,
          }
        }
      />
      <ReplyDraftLabel
        draft={false}
      />
      <ForwardRef(Relay(Connect(UnpublishedLabel)))
        publishable={
          Object {
            "$fragmentRefs": null,
            "$refType": null,
            "createdAt": "",
            "draft": false,
            "id": "replay1",
            "private": false,
            "publishedAt": "",
            "viewerCanDelete": false,
            "viewerCanUpdate": false,
          }
        }
      />
    </div>
    <div>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="primary"
        className="mr-10 reply__update-btn mt-5"
        disabled={false}
        onClick={[Function]}
      >
        <FormattedMessage
          id="global.see"
          values={Object {}}
        />
      </Button>
    </div>
  </ListGroupItem>
  <ForwardRef(Relay(UpdateReplyModal))
    onClose={[Function]}
    reply={
      Object {
        "$fragmentRefs": null,
        "$refType": null,
        "createdAt": "",
        "draft": false,
        "id": "replay1",
        "private": false,
        "publishedAt": "",
        "viewerCanDelete": false,
        "viewerCanUpdate": false,
      }
    }
    show={false}
  />
  <ForwardRef(Relay(DeleteReplyModal))
    onClose={[Function]}
    reply={
      Object {
        "$fragmentRefs": null,
        "$refType": null,
        "createdAt": "",
        "draft": false,
        "id": "replay1",
        "private": false,
        "publishedAt": "",
        "viewerCanDelete": false,
        "viewerCanUpdate": false,
      }
    }
    show={false}
  />
</Fragment>
`;

exports[`<ReplyModalLink /> render a reply in a contribuable questionnaire 1`] = `
<Fragment>
  <ListGroupItem
    bsClass="list-group-item"
    className="reply align-items-center"
    id="reply-link-replay1"
    listItem={false}
  >
    <div>
      <FormattedMessage
        id="reply.show.link"
        values={
          Object {
            "date": <FormattedDate
              day="numeric"
              month="long"
              value={null}
              year="numeric"
            />,
            "time": <FormattedDate
              hour="numeric"
              minute="numeric"
              value={null}
            />,
          }
        }
      />
      <ReplyDraftLabel
        draft={false}
      />
      <ForwardRef(Relay(Connect(UnpublishedLabel)))
        publishable={
          Object {
            "$fragmentRefs": null,
            "$refType": null,
            "createdAt": "",
            "draft": false,
            "id": "replay1",
            "private": false,
            "publishedAt": "",
            "viewerCanDelete": true,
            "viewerCanUpdate": true,
          }
        }
      />
    </div>
    <div>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="warning"
        className="mr-10 reply__update-btn mt-5"
        disabled={false}
        onClick={[Function]}
      >
        <FormattedMessage
          id="glodal.edit"
          values={Object {}}
        />
      </Button>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="danger"
        className="reply__delete-btn mt-5"
        disabled={false}
        onClick={[Function]}
      >
        <FormattedMessage
          id="glodal.delete"
          values={Object {}}
        />
      </Button>
    </div>
  </ListGroupItem>
  <ForwardRef(Relay(UpdateReplyModal))
    onClose={[Function]}
    reply={
      Object {
        "$fragmentRefs": null,
        "$refType": null,
        "createdAt": "",
        "draft": false,
        "id": "replay1",
        "private": false,
        "publishedAt": "",
        "viewerCanDelete": true,
        "viewerCanUpdate": true,
      }
    }
    show={false}
  />
  <ForwardRef(Relay(DeleteReplyModal))
    onClose={[Function]}
    reply={
      Object {
        "$fragmentRefs": null,
        "$refType": null,
        "createdAt": "",
        "draft": false,
        "id": "replay1",
        "private": false,
        "publishedAt": "",
        "viewerCanDelete": true,
        "viewerCanUpdate": true,
      }
    }
    show={false}
  />
</Fragment>
`;
